clojure.lang.LispReader$ReaderException when running "lein check"

I am getting this error when running "lein check" on my application, but neither the stack trace nor the error message reveals where the source of the error is. The application otherwise compiles and runs without error using Leiningen.

Here is the stack trace:

```
$ lein check
Exception in thread "main" clojure.lang.LispReader$ReaderException:
java.lang.RuntimeException: Unsupported character: \\-
        at clojure.lang.LispReader.read(LispReader.java:215)
        at clojure.core$read.invoke(core.clj:3346)
        at clojure.core$read.invoke(core.clj:3344)
        at clojure.main$eval_opt.invoke(main.clj:295)
        at clojure.main$initialize.invoke(main.clj:316)
        at clojure.main$null_opt.invoke(main.clj:349)
        at clojure.main$main.doInvoke(main.clj:427)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at clojure.lang.Var.invoke(Var.java:419)
        at clojure.lang.AFn.applyToHelper(AFn.java:163)
        at clojure.lang.Var.applyTo(Var.java:532)
        at clojure.main.main(main.java:37)
Caused by: java.lang.RuntimeException: Unsupported character: \\-
        at clojure.lang.Util.runtimeException(Util.java:170)
        at clojure.lang.LispReader$CharacterReader.invoke(LispReader.java:951)
        at clojure.lang.LispReader.readDelimitedList(LispReader.java:1126)
        at clojure.lang.LispReader$ListReader.invoke(LispReader.java:962)
        at clojure.lang.LispReader.readDelimitedList(LispReader.java:1126)
        at clojure.lang.LispReader$ListReader.invoke(LispReader.java:962)
        at clojure.lang.LispReader.readDelimitedList(LispReader.java:1126)
        at clojure.lang.LispReader$VectorReader.invoke(LispReader.java:1061)
        at clojure.lang.LispReader.readDelimitedList(LispReader.java:1126)
        at clojure.lang.LispReader$ListReader.invoke(LispReader.java:962)
        at clojure.lang.LispReader.readDelimitedList(LispReader.java:1126)
        at clojure.lang.LispReader$ListReader.invoke(LispReader.java:962)
        at clojure.lang.LispReader.readDelimitedList(LispReader.java:1126)
        at clojure.lang.LispReader$ListReader.invoke(LispReader.java:962)
        at clojure.lang.LispReader.readDelimitedList(LispReader.java:1126)
        at clojure.lang.LispReader$ListReader.invoke(LispReader.java:962)
        at clojure.lang.LispReader.read(LispReader.java:180)
        ... 11 more
Failed.
```

--- a/leiningen-core/src/leiningen/core/eval.clj
+++ b/leiningen-core/src/leiningen/core/eval.clj
@@ -174,13 +174,12 @@ leiningen.core.utils/platform-nullsink instead."
             (.resurrect System/in))
           exit-value)))))
 
-;; work around java's command line handling on windows
-;; http://bit.ly/9c6biv This isn't perfect, but works for what's
-;; currently being passed; see http://www.perlmonks.org/?node_id=300286
-;; for some of the landmines involved in doing it properly
 (defn- form-string [form eval-in]
   (if (and (= (get-os) :windows) (not= :trampoline eval-in))
-    (let [s (pr-str (pr-str form))] (subs s 1 (dec (.length s))))
+    ;; On windows if a parameter is in double quotes, then all we need
+    ;; to worry about are double quotes, which we must escape by
+    ;; doubling them.
+    (string/replace (pr-str form) "\"" "\"\"")
     (pr-str form)))
 
 (defn- classpath-arg [project]

